hrms/overrides/employee_advance.py

```diff
@@ -0,0 +1,201 @@
+from hrms.hr.doctype.employee_advance.employee_advance import EmployeeAdvance
+from suite42_erp_app.overrides.custom_task import CustomTask, TaskType
+from erpnext.accounts.doctype.payment_entry.payment_entry import (
+    get_party_details,
+    get_account_details,
+)
+from suite42_erp_app.suite42_erp_app.doctype.suite42_application_config.suite42_application_config import (
+    Suite42ApplicationConfig,
+)
+from erpnext.accounts.doctype.sales_invoice.sales_invoice import get_bank_cash_account
+from suite42_erp_app.suite42_erp_app.common.authentication.role import RoleEnum
+from frappe.utils import cstr, flt
+import frappe
+from frappe import _
+
+
+class CustomEmployeeAdvance(EmployeeAdvance):
+    def validate(self):
+        self.validate_approver()
+        self.validate_advance_amount()
+        self.set_status()
+        old_doc = self.get_doc_before_save()
+
+        if old_doc is not None and old_doc.status == self.status:
+            if self.status == "Draft":
+                if frappe.session.user != self.owner:
+                    frappe.throw(_("Only Owner can edit in Draft State"))
+            elif self.status == "Pending Approval":
+                if not user_has_role(frappe.session.user, RoleEnum.EXPENSE_APPROVER1_ROLE.value):
+                    frappe.throw(_("Only the Added approver can edit the document"))
+
+    def validate_approver(self):
+        user_id = frappe.get_list(
+            "Employee",
+            filters={"employee": self.employee},
+            fields=["user_id"],
+            pluck="user_id",
+        )[0]
+        if user_id == self.approver_1:
+            frappe.throw(_("Employee and its Approver cannot be the same"))
+
+        if self.advance_amount >= RoleEnum.ADVANCE_AMOUNT.value and not user_has_role(
+            self.approver_1, RoleEnum.EXPENSE_APPROVER2_ROLE.value
+        ):
+            frappe.throw(
+                _(
+                    f"Select L2 Level Approval since the amount is greater than or equal to {RoleEnum.ADVANCE_AMOUNT.value}"
+                )
+            )
```

Comment:
check for approver1 or approver2 in else

Reply:
Done